be/src/storage/chunk_helper.cpp

```diff
@@ -274,7 +274,6 @@ Chunk* ChunkHelper::new_chunk_pooled(const Schema& schema, size_t chunk_size, bo
     for (size_t i = 0; i < schema.num_fields(); i++) {
         const FieldPtr& f = schema.field(i);
         auto column = force ? column_from_pool<true>(*f, chunk_size) : column_from_pool<false>(*f, chunk_size);
-        column->reserve(chunk_size);
```

Comment:
Columns are reserved here to avoid allocating many times when it is used later.
If this reserve is deleted, query performance will probably be affected by extra memory copy.
So could you elaborate more on the scenario that you meet?
This function may be used in the wrong way.

Reply:
**Columns** resrve function is very necessary, but the reserve of the **column** is not necessary. In my scenario, when using the vtune tool to detect memory leaks, I found that there is a memory exception in this block. For specific exceptions, please refer to the screenshot above.

Reply:
Actually, the columns will be filled by the function callers in most cases.
Suppose that the scanner call this function to get a chunk, it will fill the chunk with data read from disk.
So the memory will be allocated somewhere eventually and in a less efficient way.

Reply:
So, Is original logical currently correct?

Reply:
In most cases, it is right

Reply:
Original memory check result:
![image](https://github.com/StarRocks/starrocks/assets/40750002/8d877d3e-3b42-4468-8e2a-ba14b4fa1e3b)
![image](https://github.com/StarRocks/starrocks/assets/40750002/7ed8e851-606e-4345-8ae9-825aa8ee124e)
After apply this pr,the memory check result
![image](https://github.com/StarRocks/starrocks/assets/40750002/58e98bf0-86dd-4438-99d9-98a15109ce34)
![image](https://github.com/StarRocks/starrocks/assets/40750002/2103981f-0d30-4c2a-83df-4ee6a12a5103)
There is a serious memory leak issue. Why does not column_pool Reclaim memory which only increase without decrease.

Reply:
column_pool will be reused to accelerate other queries. you can disable it by setting `disable_column_pool=true` in be.conf

Reply:
According to actual testing, it was found that column_pool is not reused but constantly growing.

Reply:
![image](https://github.com/StarRocks/starrocks/assets/40750002/42baa176-face-4e8c-afd1-74a8c06a62e4)
Take a look at the column_pool's code again, only releasing memory occupied by 'large objects'.